src/platform/monitoring/DowntimeNotification/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,89 @@
+ * @property {string} [appTitle] - The name of the consuming application, which will be displayed in downtime messaging.
+ * @property {node} [children] - React components to be rendered based on downtime.
+ * @property {node} [content] - Alias for React.children.
+ * @property {Array<string>} dependencies - An array of services that the consuming application requires in order to operate.
+ * @property {function} getScheduledDowntime - [Provided by container] An action creator that retrieves the array of downtime from the API downtime endpoint.
+ * @property {boolean} isReady - [Provided by container] A flag for indicating whether the downtime array has been retrieved from the API and if the component can render.
+ * @property {Node} [loadingIndicator] - A React component that will be rendered while the request to the API for downtime information is pending.
+ * @property {function} [render] - A function that may be supplied for custom rendering, useful for customizing how downtime/downtime approaching is handled. Receives the derived status, downtimeWindow, downtimeMap, children as arguments.
+ * @module platorm/monitoring/DowntimeNotification
+ */
+class DowntimeNotification extends React.Component {
+
+  static propTypes = {
+    appTitle: PropTypes.string,
+    children: PropTypes.node,
+    content: PropTypes.node,
+    dependencies: PropTypes.arrayOf(PropTypes.oneOf(Object.values(services))).isRequired,
+    getScheduledDowntime: PropTypes.func.isRequired,
+    isReady: PropTypes.bool,
+    loadingIndicator: PropTypes.node,
+    render: PropTypes.func
+  };
+
+  componentDidMount() {
+    if (this.props.shouldSendRequest) this.props.getScheduledDowntime();
+  }
+
+  render() {
+    if (!this.props.isReady) {
+      return this.props.loadingIndicator || <LoadingIndicator message={`Checking the ${this.props.appTitle} status...`}/>;
+    }
+
+    const downtime = getSoonestDowntime(this.props.serviceMap, this.props.dependencies);
+    const children = this.props.children || this.props.content;
+
+    let content = children;
+
+    if (downtime) {
+      if (this.props.render) {
+        content = this.props.render(downtime, children);
+      } else if (downtime.status === serviceStatus.down) {
+        content = (
+          <Down
+            appTitle={this.props.appTitle}
+            endTime={downtime.endTime}/>
+        );
+      } else if (downtime.status === serviceStatus.downtimeApproaching) {
+        content = (
+          <DowntimeApproaching
+            appTitle={this.props.appTitle}
+            startTime={downtime.startTime}
+            endTime={downtime.endTime}>{children}</DowntimeApproaching>
+        );
+      }
+    }
+
+    return content;
+  }
+}
+
+const mapStateToProps = (state) => {
+  const shouldSendRequest = !state.scheduledDowntime.isReady && !state.scheduledDowntime.isPending;
```

Comment:
Something like `shouldRequestScheduledDowntime` or `downtimeScheduleRequested` would be clearer.

Reply:
I can see that but being in the DowntimeNotification component I feel like `shouldSendRequest` is pretty clear.